src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double KOEF_OF_GROW = 1.5;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        capacityCheck();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexCheck(index);
```

Comment:
```suggestion
        checkIndex(index);
```
(https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)